src/main/java/seedu/toluist/controller/TaskController.java

```diff
@@ -8,109 +8,79 @@
-                + "(\\s+(?<index>\\d+))?"
-                + "(\\s+(?<description>.+))?\\s*";
+    private String commandTemplate;
 
-    private static final String TASK_COMMAND = "command";
-    private static final String TASK_VIEW_INDEX = "index";
-    private static final String TASK_DESCRIPTION = "description";
+    protected static final String TASK_VIEW_INDEX = "index";
+    protected static final String TASK_DESCRIPTION = "description";
+    protected static final String TASK_START_DATE_KEYWORD = "startdate/";
+    protected static final String TASK_END_DATE_KEYWORD = "enddate/";
 
-    private static final String COMMAND_ADD_TASK = "add";
-    private static final String COMMAND_UPDATE_TASK = "update";
-    private static final String COMMAND_DELETE_TASK = "delete";
+    protected static final int START_INDEX = 0;
+    protected static final int INVALID_INDEX = -1;
 
-    private static final String RESULT_MESSAGE_ADD_TASK = "New task added";
-    private static final String RESULT_MESSAGE_UPDATE_TASK = "Task updated";
-    private static final String RESULT_MESSAGE_DELETE_TASK = "Task deleted";
-
-    private Logger logger = LogsCenter.getLogger(getClass());
+    protected Logger logger = LogsCenter.getLogger(getClass());
 
     public TaskController(Ui renderer) {
         super(renderer);
     }
 
-    public CommandResult execute(String command) {
-        logger.info(getClass().getName() + "will handle command");
-
-        TodoList todoList = TodoList.load();
-        CommandResult commandResult = new CommandResult("");
-
-        HashMap<String, String> tokens = tokenize(command);
-
-        String taskCommand = tokens.get(TASK_COMMAND);
-        String description = tokens.get(TASK_DESCRIPTION);
-        String indexToken = tokens.get(TASK_VIEW_INDEX);
-        int index = indexToken != null ? Integer.parseInt(indexToken) - 1 : -1;
-        Task task = indexToken != null
-                ? UiStore.getInstance().getTasks().get(index)
-                : null;
-
-        switch (taskCommand) {
-        case COMMAND_ADD_TASK:
-            commandResult = add(todoList, description);
-            break;
-        case COMMAND_UPDATE_TASK:
-            commandResult = update(todoList, task, description);
-            break;
-        case COMMAND_DELETE_TASK:
-            commandResult = delete(todoList, task);
-            break;
-        default:
-            break;
-        }
-
-        if (todoList.save()) {
-            uiStore.setTask(todoList.getTasks());
-            renderer.render();
-        }
-
-
-        return commandResult;
+    public TaskController(Ui renderer, String commandTemplate) {
+        super(renderer);
+        this.commandTemplate = commandTemplate;
     }
 
-    @Override
-    public HashMap<String, String> tokenize(String commandArgs) {
-        Pattern pattern = Pattern.compile(COMMAND_TEMPLATE);
+    public HashMap<String, String> tokenize(String commandArgs, Boolean hasIndex, Boolean hasDescription) {
```

Comment:
Notice how the `TaskController` is only actually implementing the `tokenize` method. The children classes are not actually using anything from the parent. In this case, using composition is more appropriate. Try to take this out into a `TaskTokenizer` class which the `tokenize` method in each of the task controllers can then use